Makefile

```diff
@@ -374,6 +384,19 @@ unit-test: envtest
 	OPERATOR_NAMESPACE=$(OPERATOR_NAMESPACE) KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) --bin-dir $(LOCALBIN) -p path)" go test $(TEST_SRC) -v  -coverprofile cover.out
 CLEANFILES += cover.out
 
+$(PROMETHEUS_TEST_DIR)/%.rules.yaml: $(PROMETHEUS_TEST_DIR)/%.unit-tests.yaml $(PROMETHEUS_CONFIG_YAML) $(YQ)
+	$(YQ) eval ".data.\"$(@F:.rules.yaml=.rules)\"" $(PROMETHEUS_CONFIG_YAML) > $@
+
+# Run prometheus-alert-unit-tests
+.PHONY: test-alerts
+test-alerts: $(PROMETHEUS_ALERT_TESTS:.unit-tests.yaml=.rules.yaml)
```

Comment:
AFAIR this syntax is considered deprecated in favour of $(patsubst) in gnu make. I personally do not have a strong opinion (especially for the rule above). But the value is used in 3 places, makes sense to move it in a variable as well.

Also PROMETHEUS_ALERT_TESTS is sort of temporary variable, not something high level "user facing" like directories but produced from them, so I think it is better to keep it close to the rule. It also fixes small issue that with such a logic it's not obvious that `$(PROMETHEUS_TEST_DIR)/%.rules.yaml` rule produces prerequisites for `test-alerts` rule  (PROMETHEUS_TEST_DIR name is lost producing PROMETHEUS_ALERT_TESTS somewhere). Making a variable with `*.rules.yaml` will contribute to the clarity as well.

And back to naming. I'm still confused with the `unit` part in the names. Yes, I understand, that we can call every test as unit test, it depends of size of the unit :). But even you work fine with PROMETHEUS_ALERT_TESTS instead of PROMETHEUS_ALERT_UNIT_TESTS which it provides based on naming.

Reply:
I can't find anything to say that this syntax (substitution references) is deprecated :thinking: The text functions doc page says:
> Substitution references (see [Substitution References](https://www.gnu.org/software/make/manual/html_node/Substitution-Refs.html)) are a simpler way to get the effect of the patsubst function
then that substitution references page says that:
> We provide substitution references as well as patsubst for compatibility with other implementations of make.
which is the closest I found, but not specified as deprecated explicitly. I don't have a strong opinion of this changing, if we prefer using the patsubst function.

I agree that specifying this as a variable is a good idea :+1: @biswassri could you please do that?

Reply:
> I can't find anything to say that this syntax (substitution references) is deprecated 🤔 The text functions doc page says:
> 
> > Substitution references (see [Substitution References](https://www.gnu.org/software/make/manual/html_node/Substitution-Refs.html)) 
[...]

Probably that wording `We provide substitution references as well as patsubst for compatibility with other implementations of make.` confused me. Sounds as native gnu make ways is patsubst. But as I said, I do not have strong opinion, so substitution reference is fine for me.

Reply:
Same, no strong opinion either way from me :+1:

Reply:
@grdryn @ykaliuta Any objection to leave it as is?

Reply:
> @grdryn @ykaliuta Any objection to leave it as is?

No objections from me to leave substitution references, but yes to repeat it in 3 places.

Reply:
Sure, moving this to a variable.